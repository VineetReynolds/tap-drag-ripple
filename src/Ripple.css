.btn {
    margin: 50px auto;
    border-radius: 25px;
    background-color: #ffffff;
    box-shadow: 0 2px 4px 0 #888888;
    display: inline-block;
    padding: 15px 50px;
    color: #333333;
    width: 15em;
    height: 10em;
  }
  .center {
    text-align: center;
  }
  .ripple {
    position: relative;
    overflow: hidden;
  }
  .ripple .rippleContainer {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
  }
  .ripple .rippleContainer span {
    /* transform: scale(0); */
    width: 50px;
    height: 50px;
    border-radius: 100%;
    /* border-radius: 100%; */
    position: absolute;
    opacity: 0.25;
    background-color: #5300e8;
    animation: ripple 1000ms;
    border-style: solid;
  }
  
  @keyframes ripple {
    to {
      opacity: 0;
      transform: scale(2);
    }
  }

  .block {
      background-color: #333333;
      width: 100%;
      height: 2em;
      margin-top: 0;
  }