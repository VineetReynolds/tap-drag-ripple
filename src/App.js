import React from 'react';
import logo from './logo.svg';
import './App.css';
import Ripple from './Ripple.js';

function App() {
  return (
  <div className="center">
    <Ripple classes="btn">
      <div className="block"></div>
      Click Me
    </Ripple>
  </div>
  );
}

export default App;
