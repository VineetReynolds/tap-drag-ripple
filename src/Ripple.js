import React from 'react';
import './Ripple.css';

class Ripple extends React.Component {
    constructor(props, context) {
        super(props, context);
        console.log(this);
        this.callCleanUp = this.callCleanUp.bind(this);
        this.cleanUp = this.cleanUp.bind(this);
    };

    initializeState = () => {
      return {
        spanStyles: {},
        count: 0
      }
    }
    state = this.initializeState();
  
    /* Debounce Code to call the Ripple removing function */
    callCleanUp = (cleanup, delay) => {
        // let func = function() {
        //   clearTimeout(this.bounce);
        //   this.bounce = setTimeout(() => {
        //     cleanup();
        //   }, delay);
        // };
        // func = func.bind(this);
        // return func;
        setTimeout(() => {
            cleanup();
        }, delay);
    }
  
    showRipple = (e) => {
      const rippleContainer = e.currentTarget;
      const size = 25; //rippleContainer.offsetWidth / 2;
      const pos = rippleContainer.getBoundingClientRect();
      const x = e.pageX - pos.x - (size / 2);
      const y = e.pageY - pos.y - (size / 2);
  
      const spanStyles = { top: y + 'px', left: x + 'px', height: size + 'px', width: size + 'px', opacity: 0.4 };
      const count = this.state.count + 1;
      this.setState({
        spanStyles: {...this.state.spanStyles, [count] : spanStyles},
        count: count
      });
    }
  
    cleanUp = () =>{
        setTimeout(() => {
            const initialState = this.initializeState();
            this.setState({ ...initialState });
        }, 500);
    }
  
    renderRippleSpan = () => {
      const {showRipple = false, spanStyles = {}} = this.state;
      const spanArray = Object.keys(spanStyles);
      if (spanArray && spanArray.length > 0) {
        return (
          spanArray.map((key, index) => {
            return <span draggable key={'spanCount_' + index} className="" style={{ ...spanStyles[key]}}></span>
          })
        )
      } else {
        return null;
      }
    }
  
    render() {
      const {children= null, classes = "", onClickHandler = null} = this.props;
      return (
        <div className={'ripple ' + classes} onClick={onClickHandler}>
          {children}
          <div className="rippleContainer" onTouchStart={this.showRipple} onMouseDown={this.showRipple} onMouseUp={this.cleanUp} onTouchEnd={this.cleanUp}>
            {this.renderRippleSpan()}
          </div>
        </div>
      );
    }
  }

export default Ripple;