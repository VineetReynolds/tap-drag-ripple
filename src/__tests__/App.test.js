import React from 'react';
import {App} from '../App';
import { create } from "react-test-renderer";
import Ripple from '../Ripple';
import { render } from 'react-dom'
import { mount, shallow } from 'enzyme';

describe('App', () => {
    test('it matches snapshot', () => {
        const app = create(<Ripple></Ripple>);
        expect(app.toJSON()).toMatchSnapshot();
    })

    test('it handles click events', () => {
        const app = document.createElement('div');
        app.id = 'app';
        document.body.appendChild(app);
        const ripple = render(<Ripple></Ripple>, app);
        //console.log(ripple.find('.rippleContainer').get(0).style);
        //app.find('.rippleContainer').simulate('click');
        const root = app.firstElementChild;
        //expect(root.innerHTML).toBe('Click');
        app.dispatchEvent(new MouseEvent('click', { bubbles: true }))
        expect(app.innerHTML).toBe('Click');
    })
})