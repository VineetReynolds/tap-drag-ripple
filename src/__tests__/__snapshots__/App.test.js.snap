// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it matches snapshot 1`] = `
<div
  className="ripple "
  onClick={null}
>
  <div
    className="rippleContainer"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  />
</div>
`;
